iap/refresh/js/poll.js

```diff
@@ -0,0 +1,68 @@
+// Copyright Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+(function poll() {
+  setTimeout(function() {
+    var statusText = document.getElementById('status');
+    statusText.innerHTML = 'Polling';
+    var xhr = new XMLHttpRequest();
+    xhr.open('GET', '/status');
+    xhr.onload = function(){
+      if (xhr.status === 200) {
+        statusText.innerHTML = xhr.response;
+      } else
+      // [START handle_error]
+      if (xhr.status === 401) {
+        statusText.innerHTML = 'Login stale. <input type="button" value="Refresh" onclick="session_refresh_clicked();"/>';
+      }
+      // [END handle_error]
+      else {
+        statusText.innerHTML = xhr.statusText;
+      }
+      poll();
```

Comment:
Looks like `poll()` is always called. Maybe you could use [setInterval](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setInterval) instead of `setTimeout` in this case?

Reply:
Done